Xamarin.Forms.Platform.UAP/PlatformUWP.cs

```diff
@@ -83,7 +84,32 @@ static void OnPageActionSheet(object sender, ActionSheetArguments options)
 					options.SetResult(null);
 			};
 
-			actionSheet.ShowAt(((Page)sender).GetOrCreateRenderer().ContainerElement);
+			var pageElement = ((Page)sender).GetOrCreateRenderer().ContainerElement;
+			// If pageElement is MainPage it can have pop-up dialogs which overlap this page
+			// In order for ActionSheet to be on top, we need to show it at the dialog box
+			if (pageElement.RenderSize.Height == Window.Current.Bounds.Height
```

Comment:
Rather than comparing sizes, could you just check `(Page)sender == Application.Current.MainPage`?

Reply:
This will be true only if it is called directly from the `Application.Current.MainPage.DisplayActionSheet`. But it can also be called from any other page that fully overlaps the main page, like as my test page `<page>.DisplayActionSheet`. 
This behavior will be Ok?